x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -435,6 +444,15 @@ private ByteRange mapSubRangeToRegion(ByteRange range, int region) {
         );
     }
 
+    // package private for tests
+    int computeCacheFileRegionSize(long fileLength, int region) {
+        if (useFullRegionSize) {
```

Comment:
I initially wanted to avoid extending the shared blob cache service class just to overload that method, but maybe I've been a bit lazy. I'll give it a try in the PR that uses this feature.

Reply:
See latest code changes for the protected method solution.